app.rb

```diff
@@ -15,19 +15,26 @@ def initialize
 
   def list_all_books
     puts "\nList of Books. \n"
-    @books.each { |book| puts "title: #{book.title} by author : #{book.author}" }
+    @books.each(&:print)
     puts "\n"
   end
 
   def list_all_people
     puts "\n"
-    @persons.each do |person|
-      if person.instance_of?(Teacher)
-        puts "[Teacher]Age: #{person.age} Name: #{person.name}, ID: #{person.id}"
-      else
-        puts "[Student]Age: #{person.age},Name: #{person.name}, ID: #{person.id} "
-      end
-    end
+    @persons.each(&:print)
+
+    puts "\n"
+  end
+
+  def list_all_rentals
+    print "\nID of person"
+    input_person_id = gets.chomp.to_i
+
+    person = @persons.find { |p| p.id == input_person_id }
+    rentals = @rentals.select { |r| r.person == person }
+
+    puts 'Rentals'
+    rentals.each(&:print)
     puts "\n"
   end
 
```

Comment:
![image](https://github.com/erickma1/School_Library/assets/83042742/13a0a435-cd48-4785-b9ef-905d133f0172)
- [ ] I really appreciate the manager classes you've created separating the concerns, but you've missed one small requirement according to the single responsibility principle It's best if all requests for input and input validation are placed in separate classes/methods whose responsibilities are to request or validate input. Refer to this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for further guidance.

- [ ]  I also suggest you apply this approach to other methods where you're following similar patterns so please carefully investigate and refactor your code accordingly.